TYPO3.Flow/Classes/TYPO3/Flow/Object/Proxy/Compiler.php

```diff
@@ -141,6 +141,9 @@ public function getProxyClass($fullClassName)
         if (in_array(substr($fullClassName, 0, 14), $this->blacklistedSubPackages)) {
             return false;
         }
+        if ($classReflection->isFinal() && preg_match('/^\s?\*\s?\@Annotation\s/m', $classReflection->getDocComment()) === 1) {
```

Comment:
> Now is the chance for a breaking change

Mh, only because we _can_ break the API we should still try to avoid it by all means

> implementing such specific checks to avoid that is a hack IMHO

I agree that it is a bit "crude", especially because we can't use the `AnnotationParser` for that part. But I wouldn't consider it a hack. Especially if you look at the surrounding code of the Proxy building: There are lots of weird hard-coded specifics, like:

> Applying the docblock to the proxy could also have unintended results

We already do that, but we do it "selectively" for some reason: https://github.com/neos/flow-development-collection/blob/master/TYPO3.Flow/Classes/TYPO3/Flow/Object/Proxy/ProxyClass.php#L242-L256
We just skip `@Annotation` because it will *not* be returned by `ReflectionService::getClassAnnotations()`.
IMO it would be less of a hack to just copy the whole doc block